core/tx_pool.go

```diff
@@ -680,6 +685,29 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (bool, error) {
 			pool.removeTx(tx.Hash(), false)
 		}
 	}
+	// If there isn't enough free size in pool, discard underpriced transactions
+	if pool.all.TotalSize() + tx.Size() > pool.config.TotalPoolSizeLimit {
+		sizeToFree := pool.all.TotalSize() + tx.Size() - pool.config.TotalPoolSizeLimit
+		drop, freedSize := pool.priced.FreeSize(sizeToFree, tx.GasPrice(), local, pool.locals)
+		totalFreeSize := pool.config.TotalPoolSizeLimit - pool.all.TotalSize() + freedSize
+		if(totalFreeSize < tx.Size()){
+			log.Trace("Discarding underpriced transaction", "hash", hash, "price", tx.GasPrice())
+			underpricedTxCounter.Inc(1)
```

Comment:
I think I'd use a new metrics counter for this class of discards. Long term it would probably help diagnosing issues or correlating stuff. Let's make an `overflownTxCounter`, `ErrOverflown`, etc